msgraph-oauth-feature/src/com/microsoft/auth/OAuth2Feature.java

```diff
@@ -97,22 +99,20 @@ private static boolean bool(Optional<String> value) {
   static Form createTokenPayload(FeatureConfig config, Optional<String> authCode, Optional<String> refreshToken) {
     Form form = new Form();
     form.param("client_id", config.readMandatory(Property.APP_ID));
+    form.param("grant_type", grantType(config));
```

Comment:
wondering if you could implement it with a strategy? Three strategies for the three different flows?

Reply:
yes, I'll try to refactor a bit in here on Monday ... this thing is getting complexer and could benefit from it.